lib/cretonne/meta/isa/intel/encodings.py

```diff
@@ -227,6 +227,18 @@ def enc_i32_i64_ld_st(inst, w_bit, recipe, *args, **kwargs):
 enc_both(base.fill.b1, r.fiSib32, 0x8b)
 enc_both(base.regfill.b1, r.rfi32, 0x8b)
 
+# Push and Pop
+enc_i32_i64(x86.push, r.pushq, 0x50)
+enc_i32_i64(x86.pop, r.popq, 0x58)
+
+# Copy Special
+enc_i64(base.copy_special, r.copysp, 0x89)
+I32.enc(base.copy_special, *r.copysp(0x89))
```

Comment:
I gave this patch a quick try, and noticed one issue: on x64, the "mov %rsp,%rbp" is getting encoded as "0x40 0x89 0xe5", which is actually a 32-bit mov. The REX prefix is 0x40, which is missing the 'W' bit, which would make it 0x48, which is what's needed to make the operation 64-bit.

I haven't tested it, but one way to fix this might be adding a `w_bit` parameter like the one `enc_i32_i64_ld_st` has, to `enc_i64`, and then making use of it for the Copy Special encoding, though I don't know if that's the cleanest way to do it.

Reply:
The direct syntax for an encoding with a REX prefix is:

```python
I64.enc(base.copy_special, *r.copysp.rex(0x89, w=1))
```